options.go

```diff
@@ -0,0 +1,200 @@
+// If the option is None, nil is returned.
+func (o *Option[T]) Pointer() *T {
+	if o.present {
+		return &o.value
+	} else {
+		return nil
+	}
+}
+
+// Map returns a new option by applying the given function to the value of the option.
+// If the option is None, None is returned.
+func Map[A any, B any](o Option[A], f func(A) B) Option[B] {
+	if o.present {
+		return New(f(o.value))
+	} else {
+		return None[B]()
+	}
+}
+
+// FlatMap returns a new option by applying the given function to the value of the option.
+// If the option is None, None is returned.
+func FlatMap[A any, B any](o Option[A], f func(A) Option[B]) Option[B] {
+	if o.present {
+		return f(o.value)
+	} else {
+		return None[B]()
+	}
+}
+
+// String returns the string representation of the wrapped value.
+// If the option is None, an empty string is returned.
+func (o Option[T]) String() string {
+	if o.present {
+		return fmt.Sprint(o.value)
+	} else {
+		return ""
+	}
+}
+
+// GoString returns the Go representation of the option.
+func (o Option[T]) GoString() string {
+	if o.present {
+		return fmt.Sprintf("options.New(%#v)", o.value)
+	} else {
+		return fmt.Sprintf("options.None[%T]()", o.value)
+	}
+}
+
+// MarshalJSON implements the [json.Marshaler] interface.
+func (o Option[T]) MarshalJSON() ([]byte, error) {
+	return json.Marshal(o.Pointer())
+}
+
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
+func (o *Option[T]) UnmarshalJSON(bytes []byte) error {
+	var p *T
+	if err := json.Unmarshal(bytes, &p); err != nil {
+		return fmt.Errorf("Option[%T].UnmarshalJSON: %w", o.value, err)
+	}
+	*o = FromPointer(p)
+	return nil
+}
+
+// Value implements the SQL [driver.Valuer] interface.
+// See http://jmoiron.net/blog/built-in-interfaces
+func (o Option[T]) Value() (driver.Value, error) {
+	if o.present {
+		return o.value, nil
+	} else {
+		return nil, nil
+	}
+}
+
+// Scan implements the SQL [driver.Scanner] interface.
+// See http://jmoiron.net/blog/built-in-interfaces
+func (o *Option[T]) Scan(src any) error {
+	if src == nil {
+		*o = None[T]()
+		return nil
+	}
+	av, err := driver.DefaultParameterConverter.ConvertValue(src)
+	if err != nil {
+		return fmt.Errorf("Option[%T].Scan: failed to convert value from SQL driver: %w", o.value, err)
+	}
+	v, ok := av.(T)
+	if !ok {
+		return fmt.Errorf("Option[%T].Scan: failed to convert value '%#v' to type %T", o.value, v, o.value)
+	}
+	*o = New(v)
+	return nil
+}
+
+// Equal returns true if the two options are equal.
+// Usually you don't need to call this method since you can use == operator.
+// This method is provided to make Option[T] comparable by [go-cmp].
+//
+// [go-cmp]: https://github.com/google/go-cmp
+func (o Option[T]) Equal(other Option[T]) bool {
+	if o.present != other.present {
+		return false
+	}
+	if !o.present {
+		return true
+	}
+	value1 := reflect.ValueOf(o.value)
+	value2 := reflect.ValueOf(other.value)
+	return value1.Equal(value2)
+}
```

Comment:
This should be tested by using go-cmp, I think.

Reply:
Testing with go-cmp makes `options` depend on go-cmp...

We can add separate Go module and test `Equal` in the module. But I don't know we should do that.

Reply:
I added tests for go-cmp and sqlx. These tests are placed in a separate module.